src/get-started/first-java-service.md

```diff
@@ -0,0 +1,605 @@
+    <!-- todo: "There are more operations of such type: ..." — what is
+    the canonical reference on the topic? Shall we add one? -->
+
+### Service API
+
+In the previous section we have implemented operations modifying the blockchain
+state. Applications usually also need a means to _query_ data. In this section,
+we will add an operation to retrieve a vehicle entry by its ID
+from the registry. This operation will be exposed through REST API.
+
+#### Find Vehicle Service Operation
+
+First, we need to add a query operation to the Service:
+
+<!-- FIXME: Replace lines: selector with inside_block:ci-find-vehicle when
+the bug with braces is resolved: ECR-4318 -->
+<!--codeinclude-->
+[MyService.findVehicle](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+lines:138-146
+<!--/codeinclude-->
+
+Although this query method will be invoked by our code, hence the signature
+we use may be arbitrary, the signature is similar to the transaction methods:
+it takes the operation arguments and the context (here: `String` ID
+and `BlockchainData` context).
+
+#### API Controller
+
+Next, we will add a class implementing the REST API of the service.
+It will expose our "find vehicle" operation as a `GET` method.
+
+Create a new `ApiController` class:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/ApiController.java)
+block:ApiController
+<!--/codeinclude-->
+
+The `ApiController` needs a `MyService` object to query data; and
+a `Node` object to obtain the needed context: `BlockchainData`.
+It uses [Vert.x Web][vertx-web] to define the endpoints.
+
+[vertx-web]: https://vertx.io/docs/vertx-web/java/#_basic_vert_x_web_concepts
+
+!!! note
+    We encode the response in Protocol Buffers binary format for brevity.
+    The controller may encode it in any suitable format (e.g., JSON).
+
+Finally, connect the controller to the service. `MyService` already has
+an empty `createPublicApiHandlers` method, modify it to have:
+
+<!--codeinclude-->
+[MyService.createPublicApiHandlers](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+inside_block:ci-createPublicApiHandlers
+<!--/codeinclude-->
+
+!!! success
+    That's it with the service implementation! Package
+    the [_service artifact_](../glossary.md#artifact)
+    and run the integration tests:
+
+    ```shell
+    mvn verify
+    ```
+
+    and then proceed to the next section, where we will test its operation.
+
+## Test Network
+
+Let's now launch a test network in which we can see our service operation.
+Our project already has a script launching a test network with a single
+validator node: `start-testnet.sh`.
+
+Run the script:
+
+```shell
+chmod 744 start-testnet.sh # Allow the script execution, needed once
```

Comment:
This file is executable in our repository, but if the reader follows the tutorial and creates a project from the service archetype, then this script will come from its resources and won't have the executable bit (because it gets lost in the JAR resources).